modules/biobank/php/pooldao.class.inc

```diff
@@ -0,0 +1,307 @@
+        $this->db->insertOnDuplicateUpdate(
+            'biobank_pool',
+            $data['biobank_pool']
+        );
+
+        $poolId = $pool->getId();
+        if (isset($poolId)) {
+            // If update:
+            // XXX: Logic will likely eventually go here.
+        } else {
+            // If insert:
+            $poolId = $this->db->getLastInsertId();
+            // set the respective data array index to the value of the new
+            // pool ID
+            $specimenIds = $pool->getSpecimenIds();
+            foreach ($specimenIds as $id) {
+                $data['biobank_specimen_pool_rel'][$id]['PoolID'] = $poolId;
+            }
+        }
+
+        // insert or update biobank_specimen_pool_rel with relevant data.
+        foreach ($data['biobank_specimen_pool_rel'] as $insert) {
+            $this->db->insertOnDuplicateUpdate(
+                'biobank_specimen_pool_rel',
+                $insert
+            );
+        }
+
+        // XXX: decide if quantity of all pooled specimens should go to 0.
+        // XXX: logic for adding preparations to pools will likely go here as well.
+    }
+
+    /**
+     * This method takes a Pool Object and prepares the data to be inserted into
+     * the database by converting it to a data array.
+     *
+     * @param Pool $pool Pool Object to be converted
+     *
+     * @return array $data Array containing the data to be inserted
+     */
+    private function getSQLFromArray(array $pool) : array
+    {
+        $poolData = array(
+                     'PoolID'   => $pool['id'],
+                     'Label'    => $pool['label'],
+                     'CenterID' => $pool['centerId'],
+                     'Date'     => $pool['date'],
+                     'Time'     => $pool['time'],
+                    );
+
+        $specimenIds = $pool->getSpecimenIds();
+        foreach ($specimenIds as $id) {
+            $specimenPoolData[$id] = array(
+                                      'PoolID'     => $pool['id'],
+                                      'SpecimenID' => $id,
+                                     );
+        }
+
+        return array(
+                'biobank_pool'              => $poolData,
+                'biobank_specimen_pool_rel' => $specimenPoolData,
+               );
+    }
+    
+    /**
+     * This function takes an array that resulted from an SQL query and
+     * indexes it in a format that can be used by the rest of the code.
+     *
+     * @param array $data Values to be reassigned.
+     *
+     * @return array
+     */
+    private function getArrayFromSQL(array $data, array $specimenIds) : array
+    {
+        return array(
+                'id'          => intval($data['PoolID']),
+                'label'       => $data['Label'],
+                'specimenIds' => $specimenIds,
+                'centerId'    => intval($data['CenterID']),
+                'date'        => $data['Date'],
```

Comment:
Maybe use DateTime objects here if it makes sense

Reply:
Will look into it!

Reply:
I've implemented datetime, but i'm not sure if it should be instantiated here. My intuition is to instantiate and stringify the datetime objects as the date and time are being set to the entity object in the toInstanceFromArray and toArrayFromInstance functions.

@johnsaigle @ridz1208 thoughts?

Reply:
I have gone ahead and used datetime object when instantiating or dissolving the class. If there are any issues they can be addressed in the following commit.